src/Microsoft.Graph.Core/Requests/GraphClientFactory.cs

```diff
@@ -0,0 +1,259 @@
+        }
+        /// <summary>
+        /// Creates a new <see cref="HttpClient"/> instance configured with the handlers provided.
+        /// </summary>
+        /// <param name="sovereignCloud">The <see cref="GraphServieCloudList"/> enum value to pass the national cloud root endpoint to client.</param>
+        /// <param name="handlers">An ordered list of <see cref="DelegatingHandler"/> instances to be invoked as an 
+        /// <see cref="HttpRequestMessage"/> travels from the <see cref="HttpClient"/> to the network and an 
+        /// <see cref="HttpResponseMessage"/> travels from the network back to <see cref="HttpClient"/>.
+        /// The handlers are invoked in a top-down fashion. That is, the first entry is invoked first for 
+        /// an outbound request message but last for an inbound response message.</param>
+        /// <returns>An <see cref="HttpClient"/> instance with the configured handlers.</returns>
+        public static HttpClient CreateClient(GraphServieCloudList sovereignCloud, params DelegatingHandler[] handlers)
+        {
+            string cloud = cloudList[(int)sovereignCloud];
+            string cloudAddress = cloud + "/" + version;
+            Uri address = new Uri(cloudAddress);
+            return CreateClient(defaultTimeout, address, null, null, handlers);
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="HttpClient"/> instance configured with the handlers, timeout, baseAddress,
+        /// cacheControlHeaderValue and proxy provided.
+        /// </summary>
+        /// <param name="timeout">A <see cref="TimeSpan"/> object to be passed to the client's Timeout property</param>
+        /// <param name="baseAddress">A <see cref="string"/> value to be setted as the client's BaseAddress property</param>
+        /// <param name="cacheControlHeaderValue">A <see cref="CacheControlHeaderValue"/> object to be passed to Client's CacheControlHeaderValue 
+        /// in DefaultRequestHeaders property.</param>
+        /// <param name="proxy">A <see cref="IWebProxy"/> object to be passed to client to configure InnderHandler's proxy property.</param>
+        /// <param name="handlers">An ordered list of <see cref="DelegatingHandler"/> instances to be invoked as an 
+        /// <see cref="HttpRequestMessage"/> travels from the <see cref="HttpClient"/> to the network and an 
+        /// <see cref="HttpResponseMessage"/> travels from the network back to <see cref="HttpClient"/>.
+        /// The handlers are invoked in a top-down fashion. That is, the first entry is invoked first for 
+        /// an outbound request message but last for an inbound response message.</param>
+        /// <returns>An <see cref="HttpClient"/> instance with the configured handlers.</returns>
+        public static HttpClient CreateClient(TimeSpan timeout, Uri baseAddress = null, CacheControlHeaderValue cacheControlHeaderValue = null, IWebProxy proxy = null, params DelegatingHandler[] handlers)
```

Comment:
I'm not sure we want this overload.  The factory should create with our defaults and if those defaults are not what a user wants then they can update the HttpClient that is returned.  Having a factory with parameters sends a bit of a mixed message considering they can be changed after the creation.